libraries/AP_GPS/AP_GPS_UBLOX.cpp

```diff
@@ -467,6 +467,10 @@ AP_GPS_UBLOX::read(void)
 				goto reset;
             }
             _payload_counter = 0;                       // prepare to receive payload
+            if (_payload_length == 0) {
```

Comment:
Why not just combine this with the other condition on line 462 and change the debug to just be "bad payload size %u" I don't see a good reason to go to the checksum in this case instead of just going back around the same way we do for a large payload.

Reply:
it depends whether we ever will want to handle a packet with zero payload. If we do then we need to allow it through. If we're sure that a zero payload packet is never going to be possible in UBX then we can treat it as an error. I'm not completely confident that u-blox won't send a zero payload reply to some query

Reply:
I'm confident that with the current set of messages we are using that will be true though, unless it's the prompt for a message, which again actually holds true for the moment as we don't want to respond to any of those.

Reply:
maybe, but the u-blox could be responding to an injected msg, or have a config that generates it.
Basically I wanted to make sure that our UBX parser could still handle any valid UBX msg. As far as I can see an empty payload is possible with the syntax of the UBX protocol, even if it isn't currently unused.
If we do a reset for empty payload then if a receiver ever does send one then we could end up dropping one or more real valid packets that follow the empty payload packet (as our reset handling is not really robust, it assumes a packet never has the two preamble bytes in the payload)